web/src/ferrostar-core.ts

```diff
@@ -0,0 +1,143 @@
+import { LitElement, html, css, unsafeCSS } from "lit";
+import { customElement, property } from "lit/decorators.js";
+import leafletStyles from "leaflet/dist/leaflet.css?inline";
+import L from "leaflet";
+import markerIconUrl from "../node_modules/leaflet/dist/images/marker-icon.png";
+import markerIconRetinaUrl from "../node_modules/leaflet/dist/images/marker-icon-2x.png";
+import markerShadowUrl from "../node_modules/leaflet/dist/images/marker-shadow.png";
+import init, { NavigationController, RouteAdapter } from "ferrostar";
+
+@customElement("ferrostar-core")
+class FerrostarCore extends LitElement {
+  @property()
+  valhallaEndpointUrl: string = "";
+
+  @property()
+  profile: string = "";
+
+  @property({ attribute: false })
+  httpClient?: Function = fetch;
+
+  // TODO: type
+  @property({ type: Object })
+  locationProvider!: any;
+
+  // TODO: type
+  @property({ type: Object })
+  costingOptions!: any;
+
+  routeAdapter: RouteAdapter | null = null;
+  map: L.Map | null = null;
+  navigationController: NavigationController | null = null;
+  currentLocationMapMarker: L.Marker | null = null;
+
+  static styles = [
+    unsafeCSS(leafletStyles),
+    css`
+      #map {
+        height: 100%;
+        width: 100%;
+      }
+    `,
+  ];
+
+  constructor() {
+    super();
+
+    // A workaround for avoiding "Illegal invocation"
+    if (this.httpClient === fetch) {
+      this.httpClient = this.httpClient.bind(window);
+    }
+
+    // A workaround for loading the marker icon images in Vite
```

Comment:
Any links to known issues? I'm fine not having this figured out for the current PR; just want to know what's outstanding for cleanup.

Reply:
This issue: https://stackoverflow.com/questions/44720448/fetch-typeerror-failed-to-execute-fetch-on-window-illegal-invocation
Maybe a quirk of Web Component or Lit.